File: programmers/1845.cpp
#include <vector>
#include <set>
using namespace std;

// 폰켓몬

int solution(vector<int> nums)
{
    int answer = 0;
    set <int> ponketmon(nums.begin(), nums.end());
    
    if(ponketmon.size() >= nums.size() / 2)
        answer = nums.size() / 2;
    else
        answer = ponketmon.size();
    
    return answer;
}

File: programmers/1845.py
## 폰켓몬

def solution(nums):
    answer = 0
    numbers = {}
    for num in nums:
        if numbers.get(num, None):
            numbers[num] += 1
        else:
            numbers[num] = 1
            
    answer = len(numbers) if len(numbers) <= len(nums) / 2 else len(nums) / 2
    
    return answer